consensus.go

```diff
@@ -178,6 +179,10 @@ type Pbft struct {
 
 	// notifier is a reference to the struct which encapsulates handling messages and timeouts
 	notifier StateNotifier
+
+	//for experimenting
+	roundTimeoutFunc  func() <-chan time.Time
```

Comment:
Could you somehow adapt and use this `RoundTimeout` callback? It's intention was to set custom delegate which holds custom logic for round timeouts calculation? I see that you are returning a channel here, therefore there might be needed some adjustments on that callback, but if `roundTimeoutFunc` is meant for timeout mocking, then I would say that we already have support for that via `RoundTimeout` callback.

```Go
type Config struct {
   ...
   // RoundTimeout is a function that calculates timeout based on a round number
   RoundTimeout RoundTimeout
   ...
}
```

Reply:
Agree with @Stefan-Ethernal. The idea of these wrappers was to remove externalities on the state machine. I would like to avoid wrappers that build on top of other wrappers.

Reply:
I like your points! It became much simpler!